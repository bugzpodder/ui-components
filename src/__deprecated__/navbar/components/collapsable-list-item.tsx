import Collapse from "@material-ui/core/Collapse";

import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import React, { ReactElement, ReactNode, useEffect, useState } from "react";
import { getListItemDataTestId } from "../util";

type Props = {
  /** Defines the text of the main item. */
  headerText: ReactNode;
  /** Defines props passed into main item's `ListItemText` component. */
  headerTextProps?: {
    [x: string]: any;
  };
  /** Defines props passed into main item's `ListItem` component. */
  headerItemProps?: {
    [x: string]: any;
  };
  /** Defines icon to be placed to the left of main item text. */
  headerIcon?: ReactElement;
  /** Defines the items to be collapsed (wrapped in a `List` component). */
  children: ReactNode;
  /** Overrides initial state (default: false) */
  isOpen?: boolean;
  /** Overrides the internal toggleList function */
  toggleList?: () => any;
};

export const CollapsableListItem: React.FC<Props> = props => {
  const {
    headerText,
    headerTextProps,
    headerItemProps,
    headerIcon,
    children,
    isOpen,
  } = props;

  const [isCurrentlyOpen, setIsCurrentlyOpen] = useState(!!isOpen);
  useEffect(() => {
    if (isOpen !== undefined) {
      setIsCurrentlyOpen(!!isOpen);
    }
  }, [isOpen]);

  const toggleList = () => {
    setIsCurrentlyOpen(isOpen => !isOpen);
  };

  const handleToggleList = props.toggleList || toggleList;
  const dataTestId = getListItemDataTestId(headerText);
  return (
    <>
      <ListItem
        button
        disableRipple
        data-testid={dataTestId}
        onClick={event => {
          handleToggleList();
          event.stopPropagation();
        }}
        {...headerItemProps}
      >
        {headerIcon && <ListItemIcon>{headerIcon}</ListItemIcon>}
        <ListItemText
          data-testid={`${dataTestId}-text`}
          primary={headerText}
          {...headerTextProps}
        />
        {isCurrentlyOpen ? <ExpandLess /> : <ExpandMore />}
      </ListItem>
      <Collapse in={isCurrentlyOpen} timeout={0} unmountOnExit>
        <List>{children}</List>
      </Collapse>
    </>
  );
};
