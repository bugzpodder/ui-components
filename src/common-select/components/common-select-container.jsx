// @flow
import AsyncSelect from "react-select/async";
import CreatableSelect from "react-select/creatable";
import React from "react";
import Select from "react-select";

type Props = {
  selectType: string,
  createMessage?: string => Node,
  defaultOptions: Array<CommonSelectOption>,
  loadOptions?: string => Promise<*>,
};

const CommonSelectTypeComponent = props => {
  const {
    createMessage, defaultOptions, selectType, ...other
  } = props;
  const { loadOptions } = other;
  if (loadOptions) {
    other.loadOptions = async (...args) => {
      try {
        return await loadOptions(...args);
      } catch (error) {
        // `react-select` silently swallows exceptions. Log exceptions to help with debugging.
        console.error(error);
        // Rethrow. Technically, `react-select` will swallow this exception...
        throw error;
      }
    };
  }
  switch (selectType) {
    case "async": {
      return (
        <AsyncSelect
          {...other}
          defaultOptions={defaultOptions}
          selectType={selectType}
        />
      );
    }
    case "creatable": {
      return (
        <CreatableSelect
          {...other}
          formatCreateLabel={createMessage}
          selectType={selectType}
        />
      );
    }
    default:
      return (
        <Select
          {...other}
          selectType={selectType}
        />
      );
  }
};

export const CommonSelectContainer = (props: Props) => {
  return (
    <div className="common-select__container">
      <CommonSelectTypeComponent {...props} />
    </div>
  );
};
