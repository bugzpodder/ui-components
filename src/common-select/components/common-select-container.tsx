import AsyncSelect from "react-select/async";
import CreatableSelect from "react-select/creatable";
import React, { ReactNode } from "react";
import Select from "react-select";
import { CommonSelectOption } from "../../types/select";

type Props = {
  selectType: string;
  createMessage?: (x0: string) => ReactNode;
  defaultOptions: CommonSelectOption[];
  loadOptions?: (x0: string) => Promise<any>;
};

const CommonSelectTypeComponent = props => {
  const { createMessage, defaultOptions, selectType, ...other } = props;
  const { loadOptions } = other;
  if (loadOptions) {
    other.loadOptions = async (...args) => {
      try {
        return await loadOptions(...args);
      } catch (error) {
        // `react-select` silently swallows exceptions. Log exceptions to help with debugging.
        console.error(error);
        // Rethrow. Technically, `react-select` will swallow this exception...
        throw error;
      }
    };
  }
  switch (selectType) {
    case "async": {
      return (
        <AsyncSelect
          {...other}
          defaultOptions={defaultOptions}
          selectType={selectType}
        />
      );
    }
    case "creatable": {
      return (
        <CreatableSelect
          {...other}
          formatCreateLabel={createMessage}
          selectType={selectType}
        />
      );
    }
    default:
      return <Select {...other} selectType={selectType} />;
  }
};

export const CommonSelectContainer: React.FC<Props> = props => {
  return (
    <div className="common-select__container">
      <CommonSelectTypeComponent {...props} />
    </div>
  );
};
