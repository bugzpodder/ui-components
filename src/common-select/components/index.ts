export {
  DropdownIndicator,
  ClearIndicator,
  MultiValueContainer,
  MultiValueRemove,
  NoOptionsMessage,
  Option,
} from "./sub-components";
export { CommonSelectContainer } from "./common-select-container";
