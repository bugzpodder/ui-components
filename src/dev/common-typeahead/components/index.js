// @flow
export {
  DropdownIndicator,
  ClearIndicator,
  MultiValueContainer,
  MultiValueRemove,
  NoOptionsMessage,
  Option,
} from "./sub-components";
export { CommonTypeaheadContainer } from "./common-typeahead-container";
