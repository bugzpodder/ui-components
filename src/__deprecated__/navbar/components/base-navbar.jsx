// @flow
import AppBar from "@material-ui/core/AppBar";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import React from "react";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import classNames from "classnames";
import moment from "moment";
import styles from "./base-navbar.module.scss";
import { BreastCancerRibbon, GrailLogo } from "../../../logos";
import { LIMS } from "@grail/lib";
import { Sidebar } from "./sidebar";

type Props = {
  /** Determines whether to put warning banner above the navigation bar. */
  isProduction?: boolean,
  /** Defines the title of the page to display. */
  title?: string,
  /** Defines the breadcrumbs to go underneath the navbar. */
  breadcrumbs?: React$Node,
  /** Defines the content aligned left on the navbar to the right of menu button. */
  left?: React$Node,
  /** Defines the content centered in the navbar to the right of menu button. */
  center?: React$Node,
  /** Defines the content aligned right in the navbar to the right of menu button. */
  right?: React$Node,
  /** Defines domain in which links should be routed. */
  domain: string,
  /** Defines links to use for external domains. */
  externalDomains: Map<string, string>,
  /** Overrieds lib/constants with items to populate the sidebar. */
  sidebarContent?: SidebarItem[],
  /** Defines the footer of the sidebar. */
  sidebarFooter?: React$Node,
  /** Defines the component to handle routing to internal links. */
  InternalLinkComponent?: React$ElementType,
  /** Defines the pathname. */
  currentPath: string,
  /** Determines what kind of sidebar to use. */
  drawerVariant?: "permanent" | "persistent" | "temporary",
  /**
   * Provides classNames to the navbar's sub-components. Options include:
   *
   *  - `appBar`: AppBar
   *
   *  - `sideBar`: Outmost Drawer for Sidebar
   *
   *  - `drawer`: div for List inside Sidebar
   */
  classes?: BaseNavbarClasses,
  /** Determines whether to open sidebar by default. */
  isSidebarOpen: boolean,
  /** Function to call when user clicks some button to open/close sidebar.  */
  toggleSidebar: Function,
};

export const BaseNavbar = (props: Props) => {
  const {
    isProduction = false,
    domain,
    title,
    breadcrumbs,
    left,
    center,
    right,
    sidebarFooter,
    isSidebarOpen,
    toggleSidebar,
    ...otherProps
  } = props;
  const { classes = {} } = props;
  const isLims = domain === LIMS;
  const isOctober = moment().month() === 9;
  return (
    <>
      <AppBar
        data-testid="navbar"
        position="sticky"
        className={classNames(classes.appBar, styles.appBar, "lims-navbar")}
      >
        {!isProduction && (
          <Typography
            data-testid="non-production-warning"
            className={styles.nonProductionWarning}
            variant="h6"
          >
            THIS IS A STAGING SERVER. DO NOT ENTER REAL DATA.
          </Typography>
        )}
        <Toolbar
          className={styles.toolbar}
          disableGutters
        >
          <div
            data-testid="navbar-left"
            className={styles.left}
          >
            <IconButton
              id="main-nav-button"
              data-testid="main-nav-button"
              onClick={toggleSidebar}
              color="inherit"
              aria-label="Menu"
            >
              <MenuIcon />
            </IconButton>
            {title && (
              <Typography
                data-testid="navbar-title"
                color="inherit"
                variant="h6"
              >
                {title}
              </Typography>
            )}
            {left}
          </div>
          <div
            data-testid="navbar-center"
            className={styles.center}
          >
            {center}
          </div>
          <div
            data-testid="navbar-right"
            className={styles.right}
          >
            {isLims && isOctober && <BreastCancerRibbon />}
            <div className={styles.brand}>
              <GrailLogo />
            </div>
            {right}
          </div>
        </Toolbar>
        {breadcrumbs}
      </AppBar>
      <Sidebar
        isOpen={isSidebarOpen}
        toggle={toggleSidebar}
        footer={sidebarFooter}
        domain={domain}
        {...otherProps}
      />
    </>
  );
};
