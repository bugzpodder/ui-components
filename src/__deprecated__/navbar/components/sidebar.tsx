import ArrowBack from "@material-ui/icons/ArrowBack";
import Divider from "@material-ui/core/Divider";
import Drawer from "@material-ui/core/Drawer";
import IconButton from "@material-ui/core/IconButton";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import React, {
  ComponentType,
  MouseEvent,
  ReactNode,
  useEffect,
  useMemo,
  useReducer,
} from "react";
import classNames from "classnames";
import pathToRegexp from "path-to-regexp";
import styles from "./sidebar.module.scss";
import { BaseNavbarClasses } from "../../../types/base-navbar";
import { CollapsableListItem } from "./collapsable-list-item";
import { ExternalLink } from "../../../link";
import { SidebarItem } from "@grailbio/lib";
import { getListItemDataTestId } from "../util";

type Props = {
  isOpen: boolean;
  toggle: (event?: MouseEvent<HTMLButtonElement>) => void;
  domain: string;
  externalDomains: Map<string, string>;
  sidebarContent: SidebarItem[];
  InternalLinkComponent?: ComponentType<any>;
  footer?: ReactNode;
  currentPath: string;
  drawerVariant?: "permanent" | "persistent" | "temporary";
  classes?: BaseNavbarClasses;
};

const TOGGLE = "TOGGLE";
const COLLAPSE_ALL = "COLLAPSE_ALL";
const CLEAR = "CLEAR";
const OPEN = "OPEN";

const reducer = (
  state,
  action: {
    [x: string]: any;
  },
) => {
  const { type } = action;
  switch (type) {
    case TOGGLE: {
      const openItems = new Set(state.openItems);
      const { index } = action;
      if (openItems.has(index)) {
        openItems.delete(index);
      } else {
        openItems.add(index);
      }
      return { openItems };
    }
    case OPEN: {
      const openItems = new Set(state.openItems);
      openItems.add(action.matchedItem);
      return { openItems };
    }
    case CLEAR:
      return { openItems: new Set([action.matchedItem]) };
    case COLLAPSE_ALL:
      return { openItems: new Set() };
    default:
      return state;
  }
};

export const Sidebar: React.FC<Props> = props => {
  const {
    sidebarContent = [],
    domain,
    currentPath,
    footer,
    isOpen,
    toggle,
    drawerVariant,
    classes = {},
  } = props;

  const matchedItem = useMemo(
    () =>
      sidebarContent.findIndex(({ children = null }) => {
        if (children == null) {
          return false;
        }
        return children.some((child: SidebarItem) => {
          const childLink: SidebarItem = child;
          const { domain: childDomain, exact = false, path } = childLink;
          if (childDomain === domain) {
            return pathToRegexp(path, [], { end: exact }).test(currentPath);
          }
          return false;
        });
      }),
    [currentPath, domain, sidebarContent],
  );

  const [state, dispatch] = useReducer(reducer, { openItems: new Set() });
  useEffect(() => {
    dispatch({ type: CLEAR, matchedItem });
  }, [matchedItem]);
  useEffect(() => {
    isOpen && dispatch({ type: OPEN, matchedItem });
  }, [isOpen, matchedItem]);

  const getLink = (item: SidebarItem, key: number, nested = false) => {
    const { exact = false, domain: itemDomain, path, name } = item;
    const { currentPath, domain, InternalLinkComponent } = props;
    const isPipelineUi =
      (path && path.startsWith("/pipeline-ui")) ||
      (currentPath && currentPath.startsWith("/pipeline-ui"));
    if (
      InternalLinkComponent !== undefined &&
      itemDomain === domain &&
      !isPipelineUi
    ) {
      const active = pathToRegexp(path, [], { end: exact }).test(currentPath);
      return (
        <ListItem
          id={`${domain}${item.path.replace(/\//g, "-")}`}
          data-testid={getListItemDataTestId(name)}
          key={key}
          component={InternalLinkComponent}
          to={item.path}
          exact={exact.toString()}
          button
          className={classNames({
            [styles.active]: active,
            [styles.nested]: nested,
          })}
          dense={nested}
          disableRipple
        >
          <ListItemText
            data-testid={`${getListItemDataTestId(name)}-text`}
            primary={name}
          />
        </ListItem>
      );
    }
    const domainString = props.externalDomains.get(itemDomain) || "";
    return (
      <ListItem
        key={key}
        data-testid={`${getListItemDataTestId(name)}`}
        // When navigating to or from pipeline-ui we want a full reload instead of react-router handling
        component={isPipelineUi ? "a" : ExternalLink}
        href={`${domainString}${path}`}
        button
        className={classNames({ [styles.nested]: nested })}
        dense={nested}
        disableRipple
        onClick={event => {
          event.stopPropagation();
        }}
      >
        <ListItemText
          data-testid={`${getListItemDataTestId(name)}-text`}
          primary={item.name}
        />
      </ListItem>
    );
  };

  const toggleCollapsableListItem = (index: number) => {
    dispatch({ type: TOGGLE, index, props });
  };

  const renderItem = (item: SidebarItem, index: number) => {
    if (item.children !== undefined) {
      const isOpen = state.openItems.has(index);
      return (
        <CollapsableListItem
          key={index}
          isOpen={isOpen}
          toggleList={() => toggleCollapsableListItem(index)}
          headerText={item.name}
        >
          {item.children.map((childItem, childIndex) =>
            getLink(childItem, index * 1000 + childIndex, true),
          )}
        </CollapsableListItem>
      );
    }
    return getLink(item, index);
  };

  const collapseAll = () => {
    dispatch({ type: COLLAPSE_ALL });
  };

  return (
    <Drawer
      id="sidebar-drawer"
      data-testid="navbar-sidebar"
      anchor="left"
      open={isOpen}
      transitionDuration={0}
      onClose={toggle}
      className={classes.sideBar}
      variant={drawerVariant}
    >
      {drawerVariant !== "persistent" && (
        <div>
          <IconButton
            id="close-sidebar"
            data-testid="navbar-sidebar-close"
            onClick={toggle}
          >
            <ArrowBack />
          </IconButton>
          <Divider />
          {/** the below collapse-all span is a hidden element used in E2E testing */}
          <span
            onClick={collapseAll}
            data-testid="collapse-all-sidebar-items"
            className={styles.collapseAllSidebarItems}
          />
        </div>
      )}
      <div
        tabIndex={0}
        data-testid="navbar-sidebar-list"
        role="button"
        onClick={event => {
          if (
            event.ctrlKey ||
            event.shiftKey ||
            event.metaKey || // apple
            (event.button && event.button === 1) // middle click, >IE9 + everyone else
          ) {
            return;
          }
          if (drawerVariant !== "persistent") {
            toggle();
          }
        }}
        className={classNames(styles.drawer, classes.drawer)}
      >
        <List className={styles.list}>{sidebarContent.map(renderItem)}</List>
      </div>
      <div data-testid="navbar-sidebar-footer" className={styles.drawerFooter}>
        <Divider />
        {footer}
      </div>
    </Drawer>
  );
};
