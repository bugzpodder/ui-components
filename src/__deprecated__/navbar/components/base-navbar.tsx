import AppBar from "@material-ui/core/AppBar";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import React, { ComponentType, MouseEvent, ReactNode } from "react";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import classNames from "classnames";
import styles from "./base-navbar.module.scss";

import { BaseNavbarClasses } from "../../../types/base-navbar";
import { Sidebar } from "./sidebar";
import { SidebarItem } from "../../../types/nav";

type Props = {
  /** Determines whether to put warning banner above the navigation bar. */
  isProduction?: boolean;
  /** Defines the title of the page to display. */
  title?: string;
  /** Defines the breadcrumbs to go underneath the navbar. */
  breadcrumbs?: ReactNode;
  /** Defines the content aligned left on the navbar to the right of menu button. */
  left?: ReactNode;
  /** Defines the content centered in the navbar to the right of menu button. */
  center?: ReactNode;
  /** Defines the content aligned right in the navbar to the right of menu button. */
  right?: ReactNode;
  /**
    The logo.
    Suggestion:
    `import { GrailLogo } from "@grail/common-private";`
    and set `logo` prop:
    `logo={<GrailLogo />}`
  */
  logo: ReactNode;
  /** Defines domain in which links should be routed. */
  domain: string;
  /** Defines links to use for external domains. */
  externalDomains: Map<string, string>;
  /** Overrieds lib/constants with items to populate the sidebar. */
  sidebarContent: SidebarItem[];
  /** Defines the footer of the sidebar. */
  sidebarFooter?: ReactNode;
  /** Defines the component to handle routing to internal links. */
  InternalLinkComponent?: ComponentType<any>;
  /** Defines the pathname. */
  currentPath: string;
  /** Determines what kind of sidebar to use. */
  drawerVariant?: "permanent" | "persistent" | "temporary";
  /**
   * Provides classNames to the navbar's sub-components. Options include:
   *
   *  - `appBar`: AppBar
   *
   *  - `sideBar`: Outmost Drawer for Sidebar
   *
   *  - `drawer`: div for List inside Sidebar
   */
  classes?: BaseNavbarClasses;
  /** Determines whether to open sidebar by default. */
  isSidebarOpen: boolean;
  /** Function to call when user clicks some button to open/close sidebar.  */
  toggleSidebar: (event: MouseEvent<HTMLButtonElement>) => void;
};

export const BaseNavbar: React.FC<Props> = props => {
  const {
    isProduction = false,
    domain,
    title,
    breadcrumbs,
    left,
    center,
    right,
    logo,
    sidebarFooter,
    isSidebarOpen,
    toggleSidebar,
    ...otherProps
  } = props;
  const { classes = {} } = props;
  return (
    <>
      <AppBar
        data-testid="navbar"
        position="sticky"
        className={classNames(classes.appBar, styles.appBar, "lims-navbar")}
      >
        {!isProduction && (
          <Typography
            data-testid="non-production-warning"
            className={styles.nonProductionWarning}
            variant="h6"
          >
            THIS IS A STAGING SERVER. DO NOT ENTER REAL DATA.
          </Typography>
        )}
        <Toolbar className={styles.toolbar} disableGutters>
          <div data-testid="navbar-left" className={styles.left}>
            <IconButton
              id="main-nav-button"
              data-testid="main-nav-button"
              onClick={toggleSidebar}
              color="inherit"
              aria-label="Menu"
            >
              <MenuIcon />
            </IconButton>
            {title && (
              <Typography
                data-testid="navbar-title"
                color="inherit"
                variant="h6"
              >
                {title}
              </Typography>
            )}
            {left}
          </div>
          <div data-testid="navbar-center" className={styles.center}>
            {center}
          </div>
          <div data-testid="navbar-right" className={styles.right}>
            <div className={styles.brand}>{logo && logo}</div>
            {right}
          </div>
        </Toolbar>
        {breadcrumbs}
      </AppBar>
      <Sidebar
        isOpen={isSidebarOpen}
        toggle={toggleSidebar}
        footer={sidebarFooter}
        domain={domain}
        {...otherProps}
      />
    </>
  );
};
