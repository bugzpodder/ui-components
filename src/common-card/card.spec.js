// @flow
import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import Button from "@material-ui/core/Button";
import { TestWrapper } from "@grail/components";
import { CommonCard } from "./index";

test("render common card", () => {
	const { container, getByTestId } = render(
		<TestWrapper>
			<CommonCard title="Test Card">This is a test</CommonCard>
		</TestWrapper>,
	);
	expect(getByTestId("card-header")).toHaveTextContent("Test Card");
	expect(getByTestId("card-content")).toHaveTextContent("This is a test");
	expect(container).toMatchSnapshot();
});

test("render common card with margin", () => {
	const { container, getByTestId } = render(
		<TestWrapper>
			<CommonCard
				title="Test Card"
				hasMargin={true}>
				This is a test
			</CommonCard>
		</TestWrapper>,
	);
	expect(getByTestId("card-header")).toHaveTextContent("Test Card");
	expect(getByTestId("card-content")).toHaveTextContent("This is a test");
	expect(container).toMatchSnapshot();
});

test("render common card with header actions", () => {
	const { container, getByTestId } = render(
		<TestWrapper>
			<CommonCard headerActions={[<Button key="0">Test Header Action Button</Button>]}>This is a test</CommonCard>
		</TestWrapper>,
	);
	expect(getByTestId("card-header")).toHaveTextContent("Test Header Action Button");
	expect(getByTestId("card-content")).toHaveTextContent("This is a test");
	expect(container).toMatchSnapshot();
});

test("render common card without title or header actions", () => {
	const { container, getByTestId } = render(
		<TestWrapper>
			<CommonCard>This is a test</CommonCard>
		</TestWrapper>,
	);
	expect(getByTestId("card-content")).toHaveTextContent("This is a test");
	expect(container).toMatchSnapshot();
});

test("render common card with footer actions", () => {
	const { container, getByTestId } = render(
		<TestWrapper>
			<CommonCard
				title="Test Card"
				footerActions={[<Button key="0">Test Footer Action Button</Button>]}>
				This is a test
			</CommonCard>
		</TestWrapper>,
	);
	expect(getByTestId("card-header")).toHaveTextContent("Test Card");
	expect(getByTestId("card-content")).toHaveTextContent("This is a test");
	expect(getByTestId("card-actions")).toHaveTextContent("Test Footer Action Button");
	expect(container).toMatchSnapshot();
});
