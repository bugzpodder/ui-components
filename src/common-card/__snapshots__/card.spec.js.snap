// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common card classes and margin 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation6 MuiCard-root test-root card withMargin MuiPaper-rounded"
    data-testid="card"
  >
    <div
      class="MuiCardHeader-root test-header"
      data-testid="card-header"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title card-title test-title MuiTypography-h5 MuiTypography-displayBlock"
          data-testid="card-title"
        >
          Test Card
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader test-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          data-testid="card-subheader"
        />
      </div>
      <div
        class="MuiCardHeader-action headerActions"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Test Header Action Button
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-haspopup="true"
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="dropdown-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardContent-root test-body"
      data-testid="card-body"
    >
      This is a test
    </div>
    <div
      class="MuiCardActions-root test-footer MuiCardActions-spacing footerActions"
      data-testid="card-footer-actions"
    >
      <span>
        test
      </span>
    </div>
  </div>
</div>
`;
